public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -199,6 +202,59 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (index) {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                $scope.demand = $scope.demands[index];
+
+                $scope.demand.de_posting_date= moment().toDate();
+
+                $scope.demand.de_until = moment().add(7, 'days').toDate();
+
+            }
+
+            $scope.editRepForm = function (index) {
+
+                $scope.edit_rep = !$scope.edit_rep;
+
+                $scope.representative = $scope.buyer.re_representatives[index];
+
+            }
+
+            $scope.updateDemand = function() {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                DemandEditFactory.update({id:$scope.demands[0].bu_buyer._id, demand_id:$scope.demand._id}, $scope.demand, function(success) {
+
+                    console.log("Success");
```

Comment:
@nick22891 The user must be informed whether or not the update was successful. A dialog or a toast message would suffice here.